_posts/2017-02-13-make-virtus-dry-to-speed-up-api.md

```diff
@@ -0,0 +1,398 @@
+---
+layout: post
+title:  "Make virtus dry to speed up API"
+date:   2017-02-13 08:00:00 +0200
+author: "Artur Trzop"
+tags: ruby virtus dry-rb
+---
+
+[Virtus](https://github.com/solnic/virtus) is a popular ruby gem described as attributes on steroids for plain old ruby objects. I saw it being used in many API applications so
+when I was building API for my app I chose the virtus too. Over a year ago the [virtus gem was abandoned by its creator](https://www.reddit.com/r/ruby/comments/3sjb24/virtus_to_be_abandoned_by_its_creator/) and I learned more about [dry-rb](http://dry-rb.org) - a collection of next-generation Ruby libraries. One of the [virtus problems was performance](https://github.com/solnic/virtus/issues/287) and this article will be about it.
+
+<img src="/images/blog/posts/make-virtus-dry-to-speed-up-api/dry-virtus.jpg" style="float:right;margin-left:20px;width:400px;" alt="dry virtus" />
+
+# Virtus performance issues
+
+In 2015 I started building API for my gem knapsack_pro in order to [optimize test suite split across many CI nodes](https://knapsackpro.com). Recently I started seeing some significant difference with API performance for large users' test suites. The API response took ~500ms and sometimes event much more.
```

Comment:
my gem knapsack_pro - I'll add link to this gem. I know this link is added in next sentence, but we should add it for the first occurrence.